.github/workflows/build_develop.yml

```diff
@@ -34,28 +38,81 @@ jobs:
           SENTRY_URL: ${{ secrets.SENTRY_URL }}
           SENTRY_ORG: element
           SENTRY_PROJECT: riot-web
+          # We have to update the CSP until Cloudflare Pages supports proxying redirects
+          CSP_EXTRA_SOURCE: ${{ env.R2_PUBLIC_URL }}
 
       - run: mv dist/element-*.tar.gz dist/develop.tar.gz
 
-      # We keep the latest develop.tar.gz as the artifact uploaded later expires after 24 and requires auth to download
-      # Element Desktop's fetch script uses this tarball to fetch latest develop to build Nightlies.
-      - name: Deploy develop.tar.gz to Github Pages
-        uses: JamesIves/github-pages-deploy-action@v4
-        with:
-          folder: dist
-          single-commit: true
+      # TODO: remove this once the element-desktop fetch script is updated to grab develop.element.io/develop.tar.gz
+#      - name: Deploy develop.tar.gz to Github Pages
+#        uses: JamesIves/github-pages-deploy-action@v4
+#        with:
+#          folder: dist
+#          single-commit: true
 
       - uses: actions/upload-artifact@v3
         with:
           name: webapp
           path: dist/develop.tar.gz
           retention-days: 1
+          
+      - name: Extract webapp
+        run: |
+          mkdir _deploy
+          tar xf dist/develop.tar.gz -C _deploy --strip-components=1
+          
+      - name: Copy config
+        run: cp element.io/develop/config.json _deploy/config.json
+
+      - name: Populate 404.html
+        run: echo "404 Not Found" > _deploy/404.html
+
+      - name: Populate _headers
+        run: cp .github/cfp_headers _deploy/_headers
+
+      - name: Populate _redirects
+        run: |
+          echo "/develop.tar.gz $R2_PUBLIC_URL/develop.tar.gz 301" >> _deploy/_redirects
+          for bundle in $(aws s3 ls s3://$R2_BUCKET/bundles/ --endpoint-url $R2_URL --region=auto | awk '{print $2}'); do
```

Comment:
this is adding a redirect from CF pages to R2 for each of the bundles *currently* in R2? And not the one we're about to upload?

Reply:
Indeed, i.e all non-live bundles, though ones which already running clients may still request before their next refresh

Reply:
ok, it needs comments in the file please to explain what we're doing and why. Future readers should not have to go digging through a chain of GH issues.

Reply:
actually having read the text at the top about the CSP this makes a bit more sense. something like `# redirect requests for the develop tarball, and for historical bundles, to R2` will probably do.